### PR TITLE
Optimize isCommonPartitionFound function

### PR DESCRIPTION
**Pull request by penghuo**
1. Remove unnecessary list construct operator.
  2. Optimize readabily of isCommonPartitionFound function.
  3. Add Unit Test.

```
== NO RELEASE NOTE ==
```

### CODE REVIEW

**Review by pettyjamesm — CHANGES_REQUESTED** [review]
The code changes look good to me with one minor nit about variable names. Can you squash together the two commits and update the combined commit message to reflect the current changes?

File: presto-hive/src/main/java/com/facebook/presto/hive/HiveMaterializedViewUtils.java [hunk 1]
```diff
@@ -90,42 +90,32 @@ public static void validateMaterializedViewPartitionColumns(
...
+                                NOT_SUPPORTED,
+                                format("Materialized view %s must have at least partition to base table partition mapping for all base tables.", viewName));
+                    }
+                });
     }
 
     private static boolean isCommonPartitionFound(
-            SchemaTableName baseTable,
-            List<Column> baseTablePartitions,
+            Table table,
```
[the remark this answers is not in the document]
Comment (penghuo) [reply presto-hive/src/main/java/com/facebook/presto/hive/HiveMaterializedViewUtils.java]: Done.

**Comment by penghuo** [comment]
> The code changes look good to me with one minor nit about variable names. Can you squash together the two commits and update the combined commit message to reflect the current changes?

Done.

**Review by pettyjamesm — APPROVED** [review]

**Comment by pettyjamesm** [comment]
LGTM, merging. Thanks for your contribution @penghuo

**Pull request merged**